### PR TITLE
Web4 scripts from socialdb

### LINKED ISSUE #146: Use preloads from socialdb for Web4 frontend scripts
Today we have the script [scroll_comment_into_view.js](https://github.com/NEAR-DevHub/neardevhub-contract/blob/main/src/web4/scroll_comment_into_view.js) referenced in [src/web4/handler.rs](https://github.com/NEAR-DevHub/neardevhub-contract/blob/main/src/web4/handler.rs#L108).

The scroll_comment_into_view.js is a copy of [scripts/web4browserclient.js](https://github.com/NEAR-DevHub/neardevhub-bos/blob/main/scripts/web4browserclient.js) from the frontend repo.

From a maintenance perspective it is unfortunate to store the same content in two different repositories. There should be one source. The frontend version is also one step ahead of the file stored in the contract, since it also can support scrolling linked community announcements into the view.

I've considered to use a cdn that can fetch files directly from github repositiories, but there is a risk of caching older versions in browser clients when the same URL is used.

So after some consideration, I'm proposing using web4 preloads for this javascript file. The frontend should then deploy the javascript file to socialdb, and the web4 preload should then fetch it. This can be done the same way as we fetch [metadata in handler.rs](https://github.com/NEAR-DevHub/neardevhub-contract/blob/main/src/web4/handler.rs#L27). It also means that the frontend should write the javascript file to socialdb as part of deployment.

It would also be preferable if the script is given a dedicated URL on the web4 gateway, so that we can use it for serving iframe contents to BOS as well ( instead of having iframe source code in a string in the BOS component ).

### PR DESCRIPTION
**Pull request by petersalomonsen**
Preview URL: https://devhubweb4.testnet.page/

Notice that the served index.html is referencing `web4browserclient.js` with a `blockHeight` parameter. The block height of the latest deployed `web4browserclient.js` is obtained to a preload call to socialdb. This ensures that the browser will always load the latest version of the script, and allow local caching as long as there are no changes.

image

The javascript file, which is stored in socialdb, can then be loaded from this URL: https://devhubweb4.testnet.page/resources/devhub.near/web4browserclient.js?blockHeight=127582922

This will query socialdb for the latest block height of the javascript referenced in the web4 deployment, and reference the script in the served index.html. The script will also served through web4. Having the blockheight in the query string when referencing the script, ensure that it's always the latest script being loaded, but also enables caching in the browser as long as there are no changes. This is a good alternative to referencing a CDN where you often don't have the same opportunity to check which version is the latest, and also we can reduce the number of dependencies this way. Both when it comes to referencing the script, as we now only can use the web4 gateway, and also in deployment, where everything can be deployed to socialdb.

Resolves #146

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@petersalomonsen Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/petersalomonsen)





<details><summary>Current status: <i>executed</i></summary>

| Reviewer | Score |
|--------|--------|
| @frol  | 5 |


Your contribution is much appreciated with a final score of 5!
You have received 74 (50 base + 20 weekly bonus + 5% lifetime bonus) Sloth points for this contribution


Another weekly streak completed, well done @petersalomonsen! To keep your weekly streak and get another bonus make pull request next week! Looking forward to see you in race-of-sloths
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/petersalomonsen)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: community-factory/Cargo.toml
Comment (dj8yfo) [inline community-factory/Cargo.toml]: @petersalomonsen can you remove this section?
Until build scripts for subcontracts are properly integrated, docker builds won't work. 
Also removing it is consistent with `--no-docker` in `.github/workflows/release.yml`

File: Cargo.toml
Comment (dj8yfo) [inline Cargo.toml]: same here

File: community/Cargo.toml
Comment (dj8yfo) [inline community/Cargo.toml]: same here

File: discussions/Cargo.toml
Comment (dj8yfo) [inline discussions/Cargo.toml]: same here

File: community-factory/Cargo.toml
Comment (petersalomonsen) [reply community-factory/Cargo.toml]: reproducible build sections are removed now @dj8yfo

**Review by frol — APPROVED** [review]
@race-of-sloths score 5

**Pull request merged**
